live/liveview.go

```diff
@@ -778,6 +783,47 @@ func PageTitle(ctx context.Context, newTitle string) {
 	s.title = newTitle
 }
 
+type LiveNavType string
+
+const (
+	NavPatch    LiveNavType = "live_patch"
+	NavRedirect LiveNavType = "live_redirect"
+)
+
+// PushNav supports push patching and push redirecting from server to View
+func PushNav(ctx context.Context, typ LiveNavType, path string, params url.Values, replaceHistory bool) {
+	s := socketValue(ctx)
+	if s == nil {
+		return
+	}
+	// build new URL from existing URL and new path and params
+	url := url.URL{Path: path, RawQuery: params.Encode()}
+	to := s.url.ResolveReference(&url)
+
+	kind := "push"
+	if replaceHistory {
+		kind = "replace"
+	}
+
+	// call HandleParams if view implements ParamsHandler
+	hp, ok := s.view.(ParamsHandler)
+	if ok {
+		err := hp.HandleParams(ctx, to)
+		if err != nil {
+			// TODO better error handling
+			log.Println(err)
```

Comment:
Can we return the error and let the framework user handle?

Reply:
good call